runtime/ms-rest-azure/lib/azureEnvironment.js

```diff
@@ -124,6 +124,25 @@ var supportedEnvironments = {
     // TODO: add dns suffixes for the US government for datalake store and datalake analytics once they are defined.
     azureDataLakeStoreFileSystemEndpointSuffix: 'N/A',
     azureDataLakeAnalyticsCatalogAndJobEndpointSuffix: 'N/A'
+  },
+  AzureStack: {
+    name: 'AzureStack',
+    portalUrl: 'http://onpremportalrc.northwest3.cloudapp.azurestack.selfhost.corp.microsoft.com/',
```

Comment:
binbin why are hardcoding the URLs here? for azure stack, the URLs  will be different for each environment. You should use the metadata endpoint to retrive these values rather than hard coding it